File: uniffi_bindgen/src/bindings/swift/templates/Float64Helper.swift
fileprivate struct FfiConverterDouble: FfiConverterPrimitive {
    typealias FfiType = Double
    typealias SwiftType = Double

    static func read(from buf: inout (data: Data, offset: Data.Index)) throws -> Double {
        return try lift(readDouble(&buf))
    }

    static func write(_ value: Double, into buf: inout [UInt8]) {
        writeDouble(&buf, lower(value))
    }
}

File: uniffi_bindgen/src/bindings/kotlin/templates/Float64Helper.kt
internal object FfiConverterF64 {
    fun lift(v: Double): Double {
        return v
    }

    fun read(buf: ByteBuffer): Double {
        return buf.getDouble()
    }

    fun lower(v: Double): Double {
        return v
    }

    fun write(v: Double, buf: RustBufferBuilder) {
        buf.putDouble(v)
    }
}
